### ISSUE TITLE
indent when invoked space

### ISSUE THREAD

**Issue by yanyingwang** [zenspider/enhanced-ruby-mode#149]
Hi, guys, I find out some very weird problems with `enh-ruby-mode`, here are the examples:
![2018-07-10 17 48 04](https://user-images.githubusercontent.com/4198408/42502734-6ec9156c-8469-11e8-8540-1753ff7baa27.png)

And I already read the issues at [here](https://github.com/zenspider/enhanced-ruby-mode/issues/36) and [here](https://github.com/zenspider/enhanced-ruby-mode/issues/87). But it seems `enh-ruby-mode` doesn't have a way to recognize the `space` when handling the indent.

As my test, `ruby-mode` handle this very well in the default configuration, but `enhance-ruby-mode` has a good highlight syntax and other features.

It would be very good if someone can provide a default configuration file to make `enh-ruby-mode` handle indent just like `ruby-mode`.

And also, do we have a way to indent code like this:
![2018-07-11 11 32 56](https://user-images.githubusercontent.com/4198408/42549347-388fdd06-84fe-11e8-960f-ffb0be5103f6.png)

**Comment by zenspider** [comment]
1. Please don't post graphics as your repro.

2. for your hashes, the first one (the OK one) is the supported scenario by default. You can get what you want by using `enh-ruby-hanging-brace-deep-indent-level` set to `1`... But it won't be depending on the previous line's content at all. Just an extra space.

3. Same with `enh-ruby-hanging-paren-deep-indent-level`... but I should say, that is NOT idiomatic ruby spacing.

4. AFAIK, there is no variable to do your last example.

Basically: Everything is by number of spaces, not to try to match some token in the previous line. It's probably doable, but it isn't built in.

**Comment by zenspider** [comment]
With the latest version:

```ruby
test = { a: 1,
         b: 2,
         c: 3,
       }

Test.test(a)
    .test(b)
    .test(c)

def blah
  woot
end

# Local Variables:
# enh-ruby-hanging-brace-deep-indent-level: 1
# enh-ruby-hanging-indent-level: 4
# End:
```

**Comment by zenspider** [comment]
anything?

**Comment by zenspider** [comment]
No response for a month. Closing.
